### PR TITLE
Revert explicit save

### PR DESCRIPTION
**Pull request by arottersman**
## Overview
After our attempt at #1995 we realized warning the user when they try to leave unsaved work is a lot more involved than one would hope. We're reverting the explicit-save feature for this release. 

Connects #1997

## Testing Instructions

 * Pull, and `bundle`
 * While not logged, model an AoI, then log in — confirm a project is created and saved
 * While logged in confirm every time you make it to model a project is created/saved and any subsequent changes to it are also automatically saved

### CODE REVIEW

**Comment by rajadain** [comment]
Taking a look now.

**Comment by rajadain** [comment]
Did we need to revert back to showing the login modal on app start?

File: src/mmw/js/src/app.js [hunk 1]
```diff
@@ -112,6 +112,12 @@ var App = new Marionette.Application({
         return this._mapView._leafletMap;
     },
 
+    getUserOrShowLoginIfNotItsiEmbed: function() {
+        if (!settings.get('itsi_embed')) {
+            this.getUserOrShowLogin();
```
Comment (rajadain) [inline src/mmw/js/src/app.js]: Did we need to revert back to showing the login modal on app start?

**Comment by arottersman** [comment]
I wondered the same thing. @jfrankl thoughts?

**Comment by rajadain** [comment]
Sorry I deleted my comment, trying to make it part of the code which better keeps track of resolved comments. But here it is for the record:

> Did we need to revert back to showing the login modal on app start?

**Comment by rajadain** [comment]
> I think this is good enough for now.

What I mean is that the recommended changes aren't necessary to be added in this PR, which is good as it is.

**Comment by arottersman** [comment]
Those are good points. I'm going to merge this as is, and we can make issues if @jfrankl would like some intermediate changes in the login/save behavior for this release.

**Pull request merged**